server/datastore/mysql/hosts.go

```diff
@@ -771,33 +771,19 @@ func (ds *Datastore) MarkHostsSeen(ctx context.Context, hostIDs []uint, t time.T
 
 // SearchHosts performs a search on the hosts table using the following criteria:
 //	- Use the provided team filter.
-//	- Full-text search with the "query" argument (if query == "", then no fulltext matching is executed).
-// 	Full-text search is used even if "query" is a short or stopword.
-//	(what defines a short word is the "ft_min_word_len" VARIABLE, set to 4 by default in Fleet deployments).
+//	- Search hostname, uuid, hardware_serial, and primary_ip using LIKE (mimics ListHosts behavior)
 //	- An optional list of IDs to omit from the search.
-func (ds *Datastore) SearchHosts(ctx context.Context, filter fleet.TeamFilter, query string, omit ...uint) ([]*fleet.Host, error) {
-	var sqlb strings.Builder
-	sqlb.WriteString(`SELECT
+func (ds *Datastore) SearchHosts(ctx context.Context, filter fleet.TeamFilter, matchQuery string, omit ...uint) ([]*fleet.Host, error) {
+	query := `SELECT
 		h.*,
 		COALESCE(hst.seen_time, h.created_at) AS seen_time
 	FROM hosts h
 	LEFT JOIN host_seen_times hst
-	ON (h.id=hst.host_id) WHERE`)
+	ON (h.id=hst.host_id) WHERE TRUE `
 
 	var args []interface{}
-	if len(query) > 0 {
-		sqlb.WriteString(` (
-				MATCH (hostname, uuid) AGAINST (? IN BOOLEAN MODE)
```

Comment:
The `MATCH` operator wasn't working as expected?

Reply:
Apparently not, it wasn't filtering properly when the amount of text was small. In contrast, the other approach seems to work in every case, it might be just slightly slower... but one problem at a time.